### PR TITLE
Create heliographic map header with non-zero longitude for map center

### PR DESCRIPTION
**Pull request by STBadman**
## Allow user to create heliographic map header with custom longitude roll

This PR adds a keyword "map_center_longitude" to `sunpy.map.header_helper.make_heliographic_header` to allow the user to specify the longitude of crpix1 when creating a Carrington map header. This addresses issue [7023](https://github.com/sunpy/sunpy/issues/7023).  The keyword is required to be an astropy.Quantity with units equivalent to radians. User input values satisfying this are automatically wrapped to the range [0,360] deg in the output header. 

 As discussed in [7023](https://github.com/sunpy/sunpy/issues/7023), this extends the utility of `sunpy.map.header_helper.make_heliographic_header` to allow user created maps at an arbirtrary roll angle, rather than just the default (unchanged) which sets the map center to be 0 degrees longitude (as is the case for the Stonyhurst frame).

The PR includes adding the keyword to `sunpy.map.header_helper.make_heliographic_header` , including the astropy units decorator, and includes tests in `sunpy.map.tests.test.test_header_helper.py` that ensure the units input is correctly enforced, and that the keyword supplied by the user correctly maps and wraps to the expected header value of 'crpix1'

### CODE REVIEW

File: changelog/7143.feature.rst
Comment (ayshih) [inline changelog/7143.feature.rst:1]:
```suggestion
Add a keyword (``map_center_longitude``) to :func:`~sunpy.map.header_helper.make_heliographic_header` for centering the heliographic map at a longitude other than zero longitude.
```

File: sunpy/map/header_helper.py
Comment (ayshih) [inline sunpy/map/header_helper.py:388]:
```suggestion
def make_heliographic_header(date, observer_coordinate, shape, *, frame, projection_code="CAR",
                             map_center_longitude: u.Quantity['angle'] = 0.0*u.deg):
```
You specified the argument to have a `Quantity` as the default value, but that doesn't actually specify that the argument has to be a `Quantity` or what units are allowed.  You need to specify what unit(s) are permitted so that the input can be validated.

The `UnitsError`s that you were getting were from subsequent code trying to use the provided value, rather than from validation of the inputs to this function.

File: sunpy/map/header_helper.py
Comment (ayshih) [inline sunpy/map/header_helper.py:409]:
```suggestion
    map_center_longitude : `~astropy.units.Quantity`
        Heliographic longitude of the map center
```
There is a "center pixel" only if the map has an odd number of rows and columns.

File: sunpy/map/tests/test_header_helper.py
Comment (ayshih) [inline sunpy/map/tests/test_header_helper.py:277]:
```suggestion
    with pytest.raises(u.TypeError):
        make_heliographic_header(aia171_test_map.date, aia171_test_map.observer_coordinate, [90, 180], frame='carrington', map_center_longitude=0)

    with pytest.raises(u.UnitsError):
        make_heliographic_header(aia171_test_map.date, aia171_test_map.observer_coordinate, [90, 180], frame='carrington', map_center_longitude=0*.m)
```
With the fix above to correctly use the `quantity_input` decorator, passing something that is not a `Quantity` raises a `TypeError`.

**Review by ayshih — CHANGES_REQUESTED** [review]
Thanks for the PR!  There's some stuff that needs to be fixed

File: sunpy/map/header_helper.py
Comment (STBadman) [reply sunpy/map/header_helper.py:410]: Got it, thanks

**Comment by STBadman** [comment]
Thank you so much for @ayshih for writing the changes so explicitly.

File: sunpy/map/header_helper.py
Comment (ayshih) [inline sunpy/map/header_helper.py:389]:
```suggestion
                             map_center_longitude: u.Quantity[u.deg] = 0.0*u.deg):
```
Specifying a string makes `ruff` unhappy, so let's just specify a specific angle unit instead.  It doesn't affect the behavior at all, just the error message.

**Review by ayshih — APPROVED** [review]
Looks good to me!  We need another approval before merging

To whomever clicks the button: squash upon merge

**Review by wtbarnes — APPROVED** [review]

**Pull request merged**
